web/src/main/java/com/marklogic/hub/web/web/AbstractArtifactController.java

```diff
@@ -0,0 +1,39 @@
+package com.marklogic.hub.web.web;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.dataservices.ArtifactService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+public abstract class AbstractArtifactController {
+    @Autowired
+    private HubConfig hubConfig;
+
+    public ResponseEntity<JsonNode> getArtifacts() {
```

Comment:
I recommend making all of these methods protected so that the reader knows they're only going to be used by subclasses.